apps/wing/src/commands/run.ts

```diff
@@ -2,21 +2,53 @@ import { readdirSync, existsSync } from "fs";
 import { debug } from "debug";
 import { resolve } from "path";
 import open from "open";
+import { createConsoleApp } from "@wingconsole/app";
+import { parseNumericString } from "../util";
 
-export async function run(simfile?: string) {
-  if (!simfile) {
+/**
+ * Run options for the `run` command.
+ * This is passed from Commander to the `run` function.
+ */
+export interface RunOptions {
+  /**
+   * Preferred port number.
+   *
+   * Falls back to a random port number if necessary.
+   */
+  readonly port?: string;
+}
+
+/**
+ * Runs a Wing program in the Console.
+ * @param entrypoint The program .w entrypoint. Looks for a .w file in the current directory if not specified.
+ * @param options Run options.
+ */
+export async function run(entrypoint?: string, options?: RunOptions) {
+  if (!entrypoint) {
     const wingFiles = readdirSync(".").filter((item) => item.endsWith(".w"));
     if (wingFiles.length !== 1) {
       throw new Error("Please specify which file you want to run");
     }
-    simfile = wingFiles[0];
+    entrypoint = wingFiles[0];
   }
 
-  if (!existsSync(simfile)) {
-    throw new Error(simfile + " doesn't exist");
+  if (!existsSync(entrypoint)) {
+    throw new Error(entrypoint + " doesn't exist");
   }
 
-  simfile = resolve(simfile);
-  debug("calling wing console protocol with:" + simfile);
-  await open("wing-console://" + simfile);
+  entrypoint = resolve(entrypoint);
+  debug("opening the wing console with:" + entrypoint);
+
+  const { port } = await createConsoleApp({
```

Comment:
This stays open after the Wing CLI is done processing the command, right?

Reply:
It should stay open forever (until you Ctrl+C).